src/main/kotlin/org/rust/lang/core/psi/RsStructFiller.kt

```diff
@@ -0,0 +1,91 @@
+        fieldsToAdd: List<RsFieldDecl>
+    ): RsStructLiteralField? {
+        val forceMultiLine = structLiteral.structLiteralFieldList.isEmpty() && fieldsToAdd.size > 2
+
+        var firstAdded: RsStructLiteralField? = null
+        for (fieldDecl in fieldsToAdd) {
+            val field = specializedCreateStructLiteralField(fieldDecl) ?: continue
+            val addBefore = findPlaceToAdd(field, structLiteral.structLiteralFieldList, declaredFields)
+            val added = if (addBefore == null) {
+                ensureTrailingComma(structLiteral.structLiteralFieldList)
+                structLiteral.addBefore(field, structLiteral.rbrace) as RsStructLiteralField
+            } else {
+                val comma = structLiteral.addBefore(psiFactory.createComma(), addBefore)
+                structLiteral.addBefore(field, comma) as RsStructLiteralField
+            }
+
+            if (firstAdded == null) {
+                firstAdded = added
+            }
+        }
+
+        if (forceMultiLine) {
+            structLiteral.addAfter(psiFactory.createNewline(), structLiteral.lbrace)
+        }
+
+        return firstAdded
+    }
+
+    private fun findPlaceToAdd(
+        fieldToAdd: RsStructLiteralField,
+        existingFields: List<RsStructLiteralField>,
+        declaredFields: List<RsFieldDecl>
+    ): RsStructLiteralField? {
+        // If `fieldToAdd` is first in the original declaration, add it first
+        if (fieldToAdd.referenceName == declaredFields.firstOrNull()?.name) {
+            return existingFields.firstOrNull()
+        }
+
+        // If it was last, add last
+        if (fieldToAdd.referenceName == declaredFields.lastOrNull()?.name) {
+            return null
+        }
+
+        val pos = declaredFields.indexOfFirst { it.name == fieldToAdd.referenceName }
+        check(pos != -1)
+        val prev = declaredFields[pos - 1]
+        val next = declaredFields[pos + 1]
+        val prevIdx = existingFields.indexOfFirst { it.referenceName == prev.name }
+        val nextIdx = existingFields.indexOfFirst { it.referenceName == next.name }
+
+        // Fit between two existing fields in the same order
+        if (prevIdx != -1 && prevIdx + 1 == nextIdx) {
+            return existingFields[nextIdx]
+        }
+        // We have next field, but the order is different.
+        // It's impossible to guess the best position, so
+        // let's add to the end
+        if (nextIdx != -1) {
+            return null
+        }
+
+        if (prevIdx != -1) {
+            return existingFields.getOrNull(prevIdx + 1)
+        }
+
+        return null
+    }
+
+    private fun specializedCreateStructLiteralField(fieldDecl: RsFieldDecl): RsStructLiteralField? {
+        val fieldName = fieldDecl.name ?: return null
+        val fieldType = fieldDecl.typeReference?.type ?: return null
+        val fieldLiteral = RsDefaultValueBuilder(items, mod, psiFactory, recursive).buildFor(fieldType)
```

Comment:
`items, mod, psiFactory, recursive, this`?
Can also be a field of the current class?